### PR TITLE
(PDK-1271) Allow a transport to be wrapped and used like a device

### PR DESCRIPTION
**Pull request by da-ar**

### CODE REVIEW

File: lib/puppet/resource_api/transport.rb [hunk 1]
```diff
@@ -35,6 +35,12 @@ def connect(name, connection_info)
   end
   module_function :connect # rubocop:disable Style/AccessModifierDeclarations
 
+  def inject_device(name, transport)
+    transport_wrapper = Puppet::ResourceApi::Transport::Wrapper.new(name, transport)
+    Puppet::Util::NetworkDevice.instance_variable_set(:@current, transport_wrapper)
```
Comment (adreyer) [inline lib/puppet/resource_api/transport.rb]: I updated bolt to prefer `set_device` once that gets into puppet but I guess I forgot to push it up.

```
    if  Puppet::Util::NetworkDevice.respond_to?(:set_device)
      Puppet::Util::NetworkDevice.set_device(conn_info['type'], transport_wrapper)
    else
      Puppet::Util::NetworkDevice.instance_variable_set(:@current, transport_wrapper)
    end
```

Comment (da-ar) [reply lib/puppet/resource_api/transport.rb]: Yeah, this is exactly to code @DavidS suggested earlier today :)

File: lib/puppet/resource_api/transport.rb [hunk 2]
```diff
@@ -28,13 +28,27 @@ def list
   module_function :list # rubocop:disable Style/AccessModifierDeclarations
 
   def connect(name, connection_info)
-    validate(name, connection_info)
+    # clone the connection_info parameter, as validate may modify the value being passed in
```
Comment (da-ar) [inline lib/puppet/resource_api/transport.rb]: @DavidS can you take a look at this, the intention is to address the issue that `validate` may modify the connection_info hash.

Comment (DavidS) [reply lib/puppet/resource_api/transport.rb]: fixing the validation to not modify the `connection_info` would reduce GC pressure and compute overhead.

Comment (da-ar) [reply lib/puppet/resource_api/transport.rb]: Removed the dup and updated the way validation works. Running tests it seems that leaving unknown keys/values in place has had no negative effects.

Comment (DavidS) [reply lib/puppet/resource_api/transport.rb]: :crossed_fingers:

**Pull request merged**